### PR TITLE
Mothur tmpfiles

### PR DESCRIPTION
**Pull request by kylebittinger**
This should fix problems with temporary files noted in biocore/qiime#1935.  QIIME code will need to be adjusted to use the new `tmp_dir` argument of `mothur_classify_file` here:
https://github.com/biocore/qiime/blob/9f493d2f4ee25b0d4c81a4597c48223f2b0bb123/qiime/assign_taxonomy.py

### CODE REVIEW

File: bfillings/mothur.py [hunk 1]
```diff
@@ -496,7 +512,7 @@ def parse_mothur_assignments(lines):
 
 def mothur_classify_file(
         query_file, ref_fp, tax_fp, cutoff=None, iters=None, ksize=None,
-        output_fp=None):
+        output_fp=None, tmp_dir="/tmp"):
```
Comment (gregcaporaso) [inline bfillings/mothur.py]: @kylebittinger, this shouldn't be ``/tmp`` here, right? It also looks like ``/tmp`` is still hard-coded in other places in this file - can you do a find/replace for those?

**Comment by gregcaporaso** [comment]
Thanks @kylebittinger. It looks like what you did here was parameterize where temp files are written - this is ideal for QIIME, so we can pass QIIME's preferred temp directory in. However while we're at it we should update the default for that parameter to be ``tempfile.gettempdir()`` rather than ``/tmp``. This will then allow users to override that setting with an environment variable (see [here](https://docs.python.org/2/library/tempfile.html#tempfile.tempdir)).

Thanks for the fix! We're going to do a burrito-fillings release before QIIME 1.9.1 so we'll have this change in place. We're also going to grep the burrito-fillings code base for ``/tmp`` to globally make this change.

**Comment by kylebittinger** [comment]
I'll just add this additional fix in to the pull request, if that's OK.

**Comment by kylebittinger** [comment]
Should be done, happy to address other issues if needed.

**Comment by kylebittinger** [comment]
Should be able to merge automatically now.

**Comment by jairideout** [comment]
Thanks @kylebittinger!

**Pull request merged**